src/content/app/genome-browser/state/track-config/trackConfigSlice.ts

```diff
@@ -68,54 +98,255 @@ export const updateTrackStatesAndSave: ActionCreator<
   browserStorageService.saveTrackStates(trackStates);
 };
 
+export const getTracksStateInfo = (
+  genomeId: string,
+  trackCategories: GenomeTrackCategory[]
+) => {
+  const tracksInfo: TrackConfigs = {};
+
+  tracksInfo['gene-focus'] = {
+    showSeveralTranscripts: false,
+    showTranscriptIds: false,
+    showTrackName: false,
+    showFeatureLabel: true,
+    trackType: TrackType.GENE
+  };
+
+  trackCategories.forEach((category) => {
+    category.track_list.forEach((track) => {
+      const trackId = track.track_id.replace('track:', '');
+      const trackType = getTrackType(trackId);
+      if (trackType === TrackType.GENE) {
+        tracksInfo[trackId] = {
+          showSeveralTranscripts: false,
+          showTranscriptIds: false,
+          showTrackName: false,
+          showFeatureLabel: true,
+          trackType
+        };
+      } else {
+        tracksInfo[trackId] = {
+          showTrackName: false,
+          trackType
+        };
+      }
+    });
+  });
+  return { tracks: tracksInfo };
+};
+
+export const getTrackConfigsPerGenome = (
+  genomeId: string,
+  trackCategories: GenomeTrackCategory[]
+): TrackConfigsPerGenome => {
+  return genomeId
+    ? {
+        ...defaultTrackConfigsPerGenome,
+        ...getTracksStateInfo(genomeId, trackCategories),
+        ...getPersistentTrackConfigsPerGenome(genomeId)
+      }
+    : defaultTrackConfigsPerGenome;
+};
+
+export const getPersistentTrackConfigsPerGenome = (
+  genomeId: string
+): Partial<TrackConfigsPerGenome> => {
+  const trackConfigState =
+    browserTrackConfigStorageService.getTrackConfigState();
+  if (!genomeId || !trackConfigState[genomeId]) {
+    return {};
+  }
+  return trackConfigState[genomeId] || {};
```

Comment:
Won't this do?

```return trackConfigState[genomeId] ?? {};```

Reply:
Yes, this is probably the more appropriate syntax for expressing this idea these days.

Reply:
Looking this at the second time we don't even need a default value as that's already done in line 157. Simply `return trackConfigState[genomeId];` should do.